### PR TITLE
ci: scan for secrets with gitleaks

### PR DESCRIPTION
**Pull request by wenzowski**
even though this might already be contemplated via pre-commit...

### CODE REVIEW

File: .github/workflows/test-code.yaml [hunk 1]
```diff
@@ -84,6 +84,27 @@ jobs:
...
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Install gitleaks
+        shell: bash
+        run: |
+          set -e;
+          wget https://github.com/zricethezav/gitleaks/releases/download/v8.11.2/gitleaks_8.11.2_linux_x64.tar.gz;
+          echo '9e68a91906ce2b781b2bb785285cbfe5ad294e80ec3bb250fb019650a5248d15  gitleaks_8.11.2_linux_x64.tar.gz' > gitleaks_8.11.2_checksums.txt;
+          shasum -a 256 gitleaks_8.11.2_checksums.txt --check;
+          tar -xvf gitleaks_8.11.2_linux_x64.tar.gz;
```
Comment (AntBush) [inline .github/workflows/test-code.yaml:99]: Do we want to put this in a `.sh` file like the other scripts to stay consistent?

Comment (wenzowski) [reply .github/workflows/test-code.yaml:99]: sure, on it

File: .bin/install-gitleaks-linux-x64.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+set -xeuo pipefail;
+
+version="8.11.2";
+releases_api="https://api.github.com/repos/zricethezav/gitleaks/releases/latest";
+releases_json="$(curl -s ${releases_api})";
+
+case "$OSTYPE" in
+  darwin*)  arch="darwin_x64" ;; # arm detection? I give up
```
Comment (marcellmueller) [inline .bin/install-gitleaks-linux-x64.sh:9]: 😂

**Review by marcellmueller — APPROVED** [review]
Nice and clean bash script!

**Pull request merged**